sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala

```diff
@@ -93,8 +91,8 @@ class SQLMetric(
   def +=(v: Long): Unit = add(v)
 
   // _value may be invalid, in many cases being -1. We should not expose it to the user
-  // and instead return defaultValidValue.
-  override def value: Long = if (!isValid) defaultValidValue else _value
+  // and instead return 0.
+  override def value: Long = if (!isValid) 0 else _value
```

Comment:
can we do `if (isZero) 0 else _value`? or we are worried about negative values due to bugs?

Reply:
Hm, it seems like tests fail because 


```
```